Add AnchorJS logic for header links

lucperkins opened the pull request:
This PR uses the [AnchorJS](https://github.com/bryanbraun/anchorjs) library to more easily link to specific headers. It currently uses the **¶** character for h2 headers and the **#** character for h3 headers. I'm happy to use different characters, expand to more headers, etc.

Here's an example page chosen at random:

https://deploy-preview-10155--kubernetes-io-master-staging.netlify.com/docs/tasks/tls/certificate-rotation/

Hover over headings like "Before you begin" and "Overview" to see the anchor tags in action.

lucperkins commented:
/assign @mistyhacks @chenopis

mistyhacks commented:
I looked at the preview and it looks fine. FWIW I think we should use the link glyph (the default) and not the paragraph glyph. This is the test page with the problem on the live site: https://deploy-preview-10155--kubernetes-io-master-staging.netlify.com//docs/setup/independent/kubelet-integration/#workflow-when-using-kubeadm-join Oddly on that page, some sections have a `#` character and some have the paragraph symbol. Make sure you use `touch` so the link is always visible on touch-enabled devices that can't hover.

mistyhacks commented on static/js/script.js:
```diff
@@ -1,4 +1,18 @@
- //modal close button
+function addAnchorTags() {
```
Why can't you just use the default anchorjs call? This looks super complicated.

mistyhacks replied:
Sorry I missed that. I think we should talk about this as a SIG. My personal gut-reaction opinion is that different icons will create visual inconsistency that users might not be able to easily discern the purpose of, since to notice what level a heading is at, you need to scroll back up to its parent, etc. When people grab the link, what they want is to be able to send someone (or their own browser via a bookmark) exactly there. They don't necessarily care what level it is at. I might totally be off base, though. It is an interesting idea.

neolit123 replied:
using the same icons/symbols for all heading levels seems OK.

lucperkins replied:
@mistyhacks I agree that a single icon seems like a good default for now (I've chosen **#** as that seems to be standard in the industry). I've already made that change, which you can have a look at in the preview build. We can leave open the option of a more fine-grained system down the line.

mistyhacks replied:
Where are you seeing `#` being used? I have seen https://www.compart.com/en/unicode/U+1F517 (the default for AnchorJS) used quite heavily in multiple docsets. That said, it isn't a deal breaker for me.

lucperkins replied:
@mistyhacks Okay, that works for me!

mistyhacks replied:
One more nit: When I look at https://deploy-preview-10155--kubernetes-io-master-staging.netlify.com//docs/setup/independent/kubelet-integration/#workflow-when-using-kubeadm-join, the <code> in the title should really be rendering as mono text, not bolder than the surrounding text. It's getting the bold from `#docsContent code`. The easiest thing to do is remove the `font-family: inherit` from the following selector, and it looks better to my eyes to set the font-size just slightly smaller too:

```
#docsContent h1 code, #docsContent h2 code, #docsContent h3 code, #docsContent h4 code, #docsContent h5 code, #docsContent h6 code {
    /* font-family: inherit; */
    font-size: inherit;
    background-color: transparent;
    font-size: 85%;
}
```

neolit123 commented:
thank you, this fixes the formatting in headings.

lucperkins commented:
@mistyhacks Okay, I've updated the CSS to address the issue you've pointed and changed the icon to the AnchorJS default. Let me know if there's anything else!

mistyhacks commented on static/js/script.js:
```diff
@@ -1,4 +1,12 @@
+    visible: 'touch'
+  }
+
+  anchors.add('#docsContent h2, #docsContent h3');
```
I don't think you should stop at h3, why not go all the way down to 6? I didn't notice this before.

lucperkins replied:
@mistyhacks Ah, good call. I wrongly assumed that the header sizes stopped at h3. A simple `git grep` showed that that is quite incorrect 😄

mistyhacks commented:
/hold cancel

mistyhacks commented:
/lgtm
